tests/test_hrex_rbfe.py

```diff
@@ -129,15 +129,43 @@ def sample_and_record_rss(*args, **kwargs):
     # Initial permutation should be the identity
     np.testing.assert_array_equal(result.hrex_diagnostics.replica_idx_by_state_by_iter[0], np.arange(n_windows))
 
+    # Check that we can extract replica trajectories
+    n_atoms = result.final_result.initial_states[0].x0.shape[0]
+    rng = np.random.default_rng(2024)
+    n_atoms_subset = rng.choice(n_atoms)
+    atom_idxs = rng.choice(n_atoms, n_atoms_subset, replace=False)
+    trajs_by_replica = result.extract_trajectories_by_replica(atom_idxs)
+    assert trajs_by_replica.shape == (n_windows, md_params.n_frames, n_atoms_subset, 3)
+
+    # Check that the frame-to-frame rmsd is lower for replica trajectories versus state trajectories
+    def pairwise_rmsd(traj):
+        sds = jnp.sum(jnp.diff(traj, axis=0) ** 2, axis=(1, 2))
+        return jnp.sqrt(jnp.mean(sds))
+
+    # (states, frames)
+    trajs_by_state = np.array(
+        [[np.array(frame)[atom_idxs] for frame in state_traj.frames] for state_traj in result.trajectories]
+    )
+
+    replica_traj_rmsds = jax.vmap(pairwise_rmsd)(trajs_by_replica)
+    state_traj_rmsds = jax.vmap(pairwise_rmsd)(trajs_by_state)
+
+    # should have rmsd(replica trajectory) < rmsd(state trajectory) for all pairs (replica, state)
+    assert np.max(replica_traj_rmsds) < np.min(state_traj_rmsds)
```

Comment:
[unsure] is there an even stronger (not-flaky) _absolute_ assertion we can make here?

(`np.max(replica_traj_rmsds) < threshold` for some informative + non-flaky choice of `threshold`)

Reply:
Also unsure. I think the current assertion should catch most issues in the implementation of `extract_ligand_trajectories_by_replica` (mixing up replicas and states, off-by-one errors of the kind fixed in #1321). It seems like an absolute assertion would depend more on the details of MD, and I'm not sure this would be the right place to put it.

Reply:
Agreed -- on further thought, the relative assertion implemented seems best.